doc: release-notes: Summarize zephyr Bluetooth changes (#2649)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

The following list summarizes the most important changes inherited from upstream Zephyr:

> * Bluetooth

Reviewer: @ru-fu I didn't add a `:` in the `Kernel` section. Can we do without those? or we can clean that up later

> * Bluetooth

Reviewer 2: Better to use the colon - we can clean up the other one later.

> * Bluetooth

Author: Added `:` to Kernel

  * Support for LE Advertising Extensions has been added.
  * New application-controlled data length and PHY update APIs.
  * Legacy OOB pairing support has been added.
  * Multiple improvements to OOB data access and pairing.

>   * Multiple bug fixes and improvements

Author: Not a whole lot, will remove it.

>   * Multiple bug fixes and improvements

Reviewer 2: It's still there?

>   * Multiple bug fixes and improvements

Author: Yes, I was waiting for all questions to be answered before pushing. Now it should be gone :)

  * deprecated BT_LE_SCAN_FILTER_DUPLICATE, use BT_LE_SCAN_OPT_FILTER_DUPLICATE instead
  * deprecated BT_LE_SCAN_FILTER_WHITELIST, use BT_LE_SCAN_OPT_FILTER_WHITELIST instead
  * deprecated bt_le_scan_param::filter_dup, use bt_le_scan_param::options instead
  * deprecated bt_conn_create_le(), use bt_conn_le_create() instead
  * deprecated bt_conn_create_auto_le(), use bt_conn_le_create_auto() instead
  * deprecated bt_conn_create_slave_le(), use bt_le_adv_start() instead with
    bt_le_adv_param::peer set to the remote peers address.
  * deprecated BT_LE_ADV_* macros, use BT_GAP_ADV_* enums instead

* Bluetooth LE Controller:

>   * The Controller is now 5.2 compliant.
>   * PHY support is now configurable.
>   * Only control procedures supported by the peer are now used.
>   * The nRF52820 IC is now supported
>   * The legacy Controller has been removed.

Reviewer 2: ```suggestion
  * Updated the Controller to be 5.2 compliant.
  * Made PHY support configurable.
  * Updated the Controller to only use control procedures supported by the peer.
  * Added support for the nRF52820 SoC.
  * Remove the legacy Controller.
```

Author: @ru-fu I suppose the last one should be "removed", right?

* Bluetooth Mesh

>   * The net_idx parameter has been removed from the Health Client model
>     APIs since it can be derived (by the stack) from the app_idx parameter

Reviewer 2: ```suggestion
  * Removed the ``net_idx`` parameter from the Health Client model APIs because it can be derived (by the stack) from the ``app_idx`` parameter.
```

  * Multiple bug fixes and improvements.
  * Deprecated ``BT_LE_SCAN_FILTER_DUPLICATE``.
    Use :cpp:enumerator:`BT_LE_SCAN_OPT_FILTER_DUPLICATE <bt_gap::BT_LE_SCAN_OPT_FILTER_DUPLICATE>` instead.
  * Deprecated ``BT_LE_SCAN_FILTER_WHITELIST``.
    Use :cpp:enumerator:`BT_LE_SCAN_OPT_FILTER_WHITELIST <bt_gap::BT_LE_SCAN_OPT_FILTER_WHITELIST>` instead.
  * Deprecated ``bt_le_scan_param::filter_dup``.
    Use ``bt_le_scan_param::options`` instead.
  * Deprecated ``bt_conn_create_le()``.
    Use :cpp:func:`bt_conn_le_create` instead.
  * Deprecated ``bt_conn_create_auto_le()``.
     Use :cpp:func:`bt_conn_le_create_auto` instead.
  * Deprecated ``bt_conn_create_slave_le()``.
    Use :cpp:func:`bt_le_adv_start` instead, with ``bt_le_adv_param::peer`` set to the remote peer's address.
  * Deprecated the ``BT_LE_ADV_*`` macros.
    Use the ``BT_GAP_ADV_*`` enums instead.

* Bluetooth LE Controller:

  * Updated the Controller to be 5.2 compliant.
  * Made PHY support configurable.
  * Updated the Controller to only use control procedures supported by the peer.
  * Added support for the nRF52820 SoC.
  * Remove the legacy Controller.

> * Bluetooth Mesh

Reviewer 2: ```suggestion
* Bluetooth Mesh:
```
